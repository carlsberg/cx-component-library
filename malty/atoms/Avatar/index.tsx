export { Avatar } from './Avatar';
export { AvatarSize } from './Avatar.types';
export type { AvatarProps } from './Avatar.types';
