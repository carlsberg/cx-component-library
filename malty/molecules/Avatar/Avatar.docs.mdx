---
labels: ['react', 'typescript', 'avatar']
description: 'An Avatar component.'
---
