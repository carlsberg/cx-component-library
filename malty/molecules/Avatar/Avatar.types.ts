import { SpaceProps } from '@carlsberg/malty.utils.space';

export interface AvatarProps extends SpaceProps {
  profileImg?: string;
  userName?: string;
  size?: AvatarSize;
  editable?: boolean;
  loading?: boolean;
  onClick?: () => void;
  dataQaId?: string;
}

export enum AvatarSize {
  Small = 'Small',
  Medium = 'Medium',
  Large = 'Large',
  XLarge = 'XLarge'
}

export type StyledAvatarProps = {
  profileImg?: string;
  size?: string;
  isLoading: boolean;
  editable: boolean;
} & SpaceProps;
