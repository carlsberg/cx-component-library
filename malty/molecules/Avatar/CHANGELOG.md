# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.2.0 (2024-05-31)

### Bug Fixes

- changed Avatar path ([#237](https://github.com/CarlsbergGBS/cx-component-library/issues/237)) ([53e2172](https://github.com/CarlsbergGBS/cx-component-library/commit/53e2172d73a74d2132ebec5ebc098db3ad700072))
- **DSM-168:** avatar color changes ([#256](https://github.com/CarlsbergGBS/cx-component-library/issues/256)) ([85dc059](https://github.com/CarlsbergGBS/cx-component-library/commit/85dc059456ca4a03e9f93556977c4d9f0488b001))
- **DSM-292:** avatar loading ([#246](https://github.com/CarlsbergGBS/cx-component-library/issues/246)) ([5c4ca79](https://github.com/CarlsbergGBS/cx-component-library/commit/5c4ca79b91d69e7a82bcdd4b7eeeeff25fe617af))
- Minor component fixes ([#265](https://github.com/CarlsbergGBS/cx-component-library/issues/265)) ([c75bd7c](https://github.com/CarlsbergGBS/cx-component-library/commit/c75bd7c926163db6efb65ab89a64d8637d7a4664))
- Remove typography provider from every component ([#383](https://github.com/CarlsbergGBS/cx-component-library/issues/383)) ([5d05680](https://github.com/CarlsbergGBS/cx-component-library/commit/5d05680e40d3da5bb3042bd8191427a70606afc3))
- reverted changes ([#396](https://github.com/CarlsbergGBS/cx-component-library/issues/396)) ([ecfcf57](https://github.com/CarlsbergGBS/cx-component-library/commit/ecfcf573863253fea16e24764b6f452c3cc4f215))
- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- **avatar:** [DSM-837] add tests ([#821](https://github.com/CarlsbergGBS/cx-component-library/issues/821)) ([00d1888](https://github.com/CarlsbergGBS/cx-component-library/commit/00d18880e55ce572d4d73df9adb57be41448a2ed))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **DSM-168:** avatar ([#227](https://github.com/CarlsbergGBS/cx-component-library/issues/227)) ([01df410](https://github.com/CarlsbergGBS/cx-component-library/commit/01df410e6b4af921a3a05f98fdb71e27a1d25c6c))
- **DSM-228:** Added rating component ([#379](https://github.com/CarlsbergGBS/cx-component-library/issues/379)) ([ab47354](https://github.com/CarlsbergGBS/cx-component-library/commit/ab47354db803fb051a1f87d66ee34966698df8bd))
- **DSM-362:** snapshots removed & unit test updates ([#307](https://github.com/CarlsbergGBS/cx-component-library/issues/307)) ([78423eb](https://github.com/CarlsbergGBS/cx-component-library/commit/78423eb5cfe04fe8a3a7af1f5af074ac6d35e235))
- **DSM-439:** Change font-family behavior ([#470](https://github.com/CarlsbergGBS/cx-component-library/issues/470)) ([ab6dad4](https://github.com/CarlsbergGBS/cx-component-library/commit/ab6dad414c42c4343de3de6abaa4cd3ff98cfca1))
- **general:** [DSM-953] refactor icons implementation ([#892](https://github.com/CarlsbergGBS/cx-component-library/issues/892)) ([08d259f](https://github.com/CarlsbergGBS/cx-component-library/commit/08d259f476ce97e9db8ac41d5eb4773eabca0d37))
- **global:** [DSM-784] add space props in all components ([#842](https://github.com/CarlsbergGBS/cx-component-library/issues/842)) ([9f03f59](https://github.com/CarlsbergGBS/cx-component-library/commit/9f03f59e7518a99abcea06bacbee989d1ddb0465))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** update storybook files to be compliant with v7 ([#970](https://github.com/CarlsbergGBS/cx-component-library/issues/970)) ([c61d4b4](https://github.com/CarlsbergGBS/cx-component-library/commit/c61d4b40a0755becf942ad3f28758a159f8c54e4))
- New Chip component and minor changes ([#239](https://github.com/CarlsbergGBS/cx-component-library/issues/239)) ([ce92ab0](https://github.com/CarlsbergGBS/cx-component-library/commit/ce92ab0543b96296952a29e2d43d0df29d4e188f))
- remove empty composition files ([f8173f4](https://github.com/CarlsbergGBS/cx-component-library/commit/f8173f4a2ecbf80bb7b6ffe848c023ae31819c2d))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
