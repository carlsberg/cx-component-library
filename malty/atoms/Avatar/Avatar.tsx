import { Icon, IconColor, IconName, IconSize } from '@carlsberggroup/malty.atoms.icon';
import { globalTheme as defaultTheme, TypographyProvider } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';
import { StyledAvatar, StyledCamera } from './Avatar.styled';
import { AvatarProps, AvatarSize } from './Avatar.types';

const displayInitials = (username: string) => {
  if (typeof username !== 'string') {
    return null;
  }
  const nameArray = username.split(' ');
  const firstName = nameArray[0];
  const firstInitial = firstName.charAt(0);

  if (nameArray.length === 1) {
    return firstInitial;
  }

  const lastName = nameArray[nameArray.length - 1];
  const lastInitial = lastName.charAt(0);

  return `${firstInitial}${lastInitial}`;
};

export const Avatar = ({ profileImg, username, size, editable = false }: AvatarProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  const [NumSize, setNumSize] = useState(theme.sizes.xl.value);
  const [iconSize, setIconSize] = useState(IconSize.MediumSmall);
  const [fontSize, setfontSize] = useState(theme.typography.desktop.text.small_bold['font-size'].value);

  useEffect(() => {
    switch (size) {
      case AvatarSize.Small: {
        setNumSize(theme.sizes.m.value);
        setfontSize(theme.typography.desktop.text.tiny_bold['font-size'].value);
        setIconSize(IconSize.ExtraSmall);
        break;
      }
      case AvatarSize.Medium: {
        setNumSize(theme.sizes.xl.value);
        setfontSize(theme.typography.desktop.text.small_bold['font-size'].value);
        setIconSize(IconSize.MediumSmall);
        break;
      }
      case AvatarSize.Large: {
        setNumSize(theme.sizes['3xl'].value);
        setfontSize(theme.typography.desktop.headline.medium['font-size'].value);
        setIconSize(IconSize.Medium);
        break;
      }
      case AvatarSize.XLarge: {
        setNumSize(theme.sizes['4xl'].value);
        setfontSize(theme.typography.desktop.headline.large['font-size'].value);
        setIconSize(IconSize.Large);
        break;
      }
      default: {
        setNumSize(theme.sizes.xl.value);
        setfontSize(theme.typography.desktop.text.small_bold['font-size'].value);
        setIconSize(IconSize.MediumSmall);
        break;
      }
    }
  }, [size, theme]);

  return (
    <TypographyProvider>
      <StyledAvatar fontSize={fontSize} size={NumSize} profileImg={profileImg} data-testid="avatar" theme={theme}>
        {editable && (
          <StyledCamera theme={theme} size={size}>
            <Icon color={IconColor.Primary} size={iconSize} name={IconName.Camera} />
          </StyledCamera>
        )}
        {!profileImg && username && <span> {displayInitials(username)} </span>}
        {!profileImg && !username && <Icon color={IconColor.Primary} size={iconSize} name={IconName.Customer} />}
      </StyledAvatar>
    </TypographyProvider>
  );
};
