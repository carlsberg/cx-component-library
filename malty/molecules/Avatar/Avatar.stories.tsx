import { generateStorybookSpacing } from '@carlsberggroup/malty.utils.space';
import { Story } from '@storybook/react';
import React from 'react';
import { Avatar as AvatarComponent } from './Avatar';
import { AvatarProps, AvatarSize } from './Avatar.types';

export default {
  title: 'Media/Avatar',
  component: AvatarComponent,
  parameters: {
    importObject: 'Avatar',
    importPath: '@carlsberggroup/malty.molecules.avatar'
  },
  argTypes: {
    profileImg: {
      description: 'The user profile image URL',
      control: {
        type: 'text'
      }
    },
    userName: {
      description: 'If profile image is not available the username initials will be displayed instead',
      control: {
        type: 'text'
      }
    },
    size: {
      description: 'Avatar size. Options are',
      options: Object.values(AvatarSize),
      table: {
        defaultValue: {
          summary: 'AvatarSize.Medium'
        }
      },
      control: {
        type: 'select'
      }
    },
    editable: {
      description: 'If true, avatar photo is editable',
      control: {
        type: 'boolean'
      }
    },
    loading: {
      description: 'If true, avatar shows loading indicator',
      control: {
        type: 'boolean'
      }
    },
    ...generateStorybookSpacing()
  }
};
const Template: Story<AvatarProps> = (args) => (
  <div style={{ width: '100px' }}>
    <AvatarComponent {...args} />
  </div>
);
export const Avatar = Template.bind({});

Avatar.args = {
  userName: 'John Doe',
  editable: false,
  loading: false
};
