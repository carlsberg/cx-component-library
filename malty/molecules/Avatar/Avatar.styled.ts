import { space } from '@carlsberggbs/malty.utils.space';
import styled, { css } from 'styled-components';
import { AvatarSize, StyledAvatarProps } from './Avatar.types';

export const StyledCamera = styled.div<{
  size?: AvatarSize;
}>`
  transition: opacity 0.25s ease-in-out;
  visibility: hidden;
  opacity: 0;
  display: ${({ size }) => (size === AvatarSize.Large || size === AvatarSize.XLarge ? 'block' : 'none')};
  position: absolute;
  box-sizing: content-box;
  background-color: ${({ theme }) => theme.colors.colours.default.white.value};
  ${({ theme }) =>
    css`
      border: ${theme.borders['border-1px--solid']['border-width'].value}
        ${theme.borders['border-1px--solid']['border-style'].value} ${theme.colors.colours.support[20].value};
    `}
  border-radius: ${({ theme }) => theme.sizes.s.value};
  width: ${({ theme, size }) => {
    if (size === AvatarSize.Large) {
      return theme.sizes.ms.value;
    }
    return theme.sizes.m.value;
  }};
  height: ${({ theme, size }) => {
    if (size === AvatarSize.Large) {
      return theme.sizes.ms.value;
    }
    return theme.sizes.m.value;
  }};

  ${({ size, theme }) =>
    size === AvatarSize.Large
      ? css`
          left: calc(50% - ${theme.sizes.ms.value} / 2 + 18px);
        `
      : css`
          left: calc(50% - ${theme.sizes.m.value} / 2 + ${theme.sizes.m.value});
        `};
  ${({ size, theme }) =>
    size === AvatarSize.Large
      ? css`
          top: calc(50% - ${theme.sizes.ms.value} / 2 - 18px);
        `
      : css`
          top: calc(50% - ${theme.sizes.m.value} / 2 - ${theme.sizes.m.value});
        `};
  svg {
    width: 100%;
    height: 100%;
  }
`;

export const StyledAvatar = styled.div<StyledAvatarProps>`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 100%;
  background-color: ${({ theme }) => theme.colors.colours.support[20].value};
  background-image: ${({ profileImg }) => `${profileImg ? `url(${profileImg})` : 'unset'}`};

  background-size: cover;
  background-repeat: no-repeat;
  transition: all 0.25s ease-in-out;
  cursor: ${({ editable }) => (editable ? 'pointer' : 'default')};
  ${({ editable, isLoading }) =>
    editable &&
    !isLoading &&
    css`
      &:hover {
        background-color: ${({ theme }) => theme.colors.colours.overlay['digital-black'][25].value};
        background-blend-mode: multiply;
        ${StyledCamera} {
          visibility: visible;
          opacity: 1;
          transition: opacity 0.25s ease-in-out;
        }
      }
    `}

  ${({ theme, size }) => {
    switch (size) {
      case AvatarSize.Small: {
        return css`
          height: ${theme.sizes.m.value};
          width: ${theme.sizes.m.value};

          & span {
            color: ${theme.colors.colours.default['digital-black'].value};
            font-family: ${theme.typography.desktop.text.tiny_bold['font-family'].value};
            font-size: ${theme.typography.desktop.text.tiny_bold['font-size'].value};
            font-weight: ${theme.typography.desktop.text.tiny_bold['font-weight'].value};
            text-transform: uppercase;
            white-space: nowrap;
          }
        `;
      }
      case AvatarSize.Large: {
        return css`
          height: ${theme.sizes['3xl'].value};
          width: ${theme.sizes['3xl'].value};

          & span {
            color: ${theme.colors.colours.default['digital-black'].value};
            font-family: ${theme.typography.desktop.headline.medium['font-family'].value};
            font-size: ${theme.typography.desktop.headline.medium['font-size'].value};
            font-weight: ${theme.typography.desktop.headline.medium['font-weight'].value};
            text-transform: uppercase;
            white-space: nowrap;
          }
        `;
      }
      case AvatarSize.XLarge: {
        return css`
          height: ${theme.sizes['4xl'].value};
          width: ${theme.sizes['4xl'].value};

          & span {
            color: ${theme.colors.colours.default['digital-black'].value};
            font-family: ${theme.typography.desktop.headline.large['font-family'].value};
            font-size: ${theme.typography.desktop.headline.large['font-size'].value};
            font-weight: ${theme.typography.desktop.headline.large['font-weight'].value};
            text-transform: uppercase;
            white-space: nowrap;
          }
        `;
      }
      default: {
        return css`
          height: ${theme.sizes.xl.value};
          width: ${theme.sizes.xl.value};

          & span {
            color: ${theme.colors.colours.default['digital-black'].value};
            font-family: ${theme.typography.desktop.text.small_bold['font-family'].value};
            font-size: ${theme.typography.desktop.text.small_bold['font-size'].value};
            font-weight: ${theme.typography.desktop.text.small_bold['font-weight'].value};
            text-transform: uppercase;
            white-space: nowrap;
          }
        `;
      }
    }
  }}

  ${space}
`;
export const StyledLoadingContainer = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.colors.colours.overlay.white[75].value};
  display: flex;
  align-items: center;
  justify-content: center;
`;
