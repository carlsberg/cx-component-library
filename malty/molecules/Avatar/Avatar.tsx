import { Icon, IconColor, IconName, IconSize } from '@carlsberggroup/malty.atoms.icon';
import { Loading, LoadingSize, LoadingStatus } from '@carlsberggroup/malty.molecules.loading';
import { globalTheme as defaultTheme, TypographyProvider } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';
import { StyledAvatar, StyledCamera, StyledLoadingContainer } from './Avatar.styled';
import { AvatarProps, AvatarSize } from './Avatar.types';

const displayInitials = (username: string) => {
  if (typeof username !== 'string') {
    return null;
  }
  const nameArray = username.split(' ');
  const firstName = nameArray[0];
  const firstInitial = firstName.charAt(0);

  if (nameArray.length === 1) {
    return firstInitial;
  }

  const lastName = nameArray.pop();
  const lastInitial = lastName?.charAt(0);

  return `${firstInitial}${lastInitial}`;
};

export const Avatar = ({
  profileImg,
  userName,
  size,
  editable = false,
  onClick = () => null,
  loading = false,
  dataQaId
}: AvatarProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  const [NumSize, setNumSize] = useState(theme.sizes.xl.value);
  const [iconSize, setIconSize] = useState(IconSize.MediumSmall);
  const [fontSize, setfontSize] = useState(theme.typography.desktop.text.small_bold['font-size'].value);

  const handleOnClick = () => {
    onClick();
  };

  useEffect(() => {
    switch (size) {
      case AvatarSize.Small: {
        setNumSize(theme.sizes.m.value);
        setfontSize(theme.typography.desktop.text.tiny_bold['font-size'].value);
        setIconSize(IconSize.ExtraSmall);
        break;
      }
      case AvatarSize.Medium: {
        setNumSize(theme.sizes.xl.value);
        setfontSize(theme.typography.desktop.text.small_bold['font-size'].value);
        setIconSize(IconSize.MediumSmall);
        break;
      }
      case AvatarSize.Large: {
        setNumSize(theme.sizes['3xl'].value);
        setfontSize(theme.typography.desktop.headline.medium['font-size'].value);
        setIconSize(IconSize.Medium);
        break;
      }
      case AvatarSize.XLarge: {
        setNumSize(theme.sizes['4xl'].value);
        setfontSize(theme.typography.desktop.headline.large['font-size'].value);
        setIconSize(IconSize.Large);
        break;
      }
      default: {
        setNumSize(theme.sizes.xl.value);
        setfontSize(theme.typography.desktop.text.small_bold['font-size'].value);
        setIconSize(IconSize.MediumSmall);
        break;
      }
    }
  }, [size, theme]);

  return (
    <TypographyProvider>
      <StyledAvatar
        onClick={handleOnClick}
        fontSize={fontSize}
        size={NumSize}
        profileImg={profileImg}
        data-testid={dataQaId}
        theme={theme}
        isLoading={loading}
        editable={editable}
      >
        {loading && (
          <StyledLoadingContainer>
            <Loading size={LoadingSize.Small} status={LoadingStatus.Pending} />
          </StyledLoadingContainer>
        )}
        {editable && (
          <StyledCamera data-testid={`${dataQaId}-camera-icon`} theme={theme} size={size}>
            <Icon color={IconColor.Support60} size={iconSize} name={IconName.Camera} />
          </StyledCamera>
        )}
        {!profileImg && userName && <span data-testid={`${dataQaId}-name`}> {displayInitials(userName)} </span>}
        {!profileImg && !userName && <Icon color={IconColor.Primary} size={iconSize} name={IconName.Customer} />}
      </StyledAvatar>
    </TypographyProvider>
  );
};
