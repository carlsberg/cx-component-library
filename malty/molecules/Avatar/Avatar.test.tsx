import { render } from '@carlsberg/malty.utils.test';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Avatar } from './Avatar';
import { AvatarSize } from './Avatar.types';

const imageSrc =
  'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=987&q=80';
const dataTestId = 'avatar';

describe('Avatar component', () => {
  it('should render with the username initials', () => {
    render(<Avatar userName="John Doe" />);
    const initialsSpan = screen.getByText('JD');

    expect(initialsSpan).toBeInTheDocument();
  });

  it('should render with first and last name initials when username has more than two words', () => {
    render(<Avatar userName="John Doe Yea" />);
    const initialsSpan = screen.getByText('JY');

    expect(initialsSpan).toBeInTheDocument();
  });

  it('should render with a profile image', () => {
    render(<Avatar profileImg={imageSrc} dataQaId={dataTestId} />);
    const avatarContainer = screen.getByTestId(dataTestId);

    expect(avatarContainer).toHaveStyle(`background-image: url(${imageSrc})`);
  });

  it('should not render username initials when profile picture is available', () => {
    render(<Avatar userName="John Doe" profileImg={imageSrc} />);

    const elementWithInitials = screen.queryByText('JD');
    expect(elementWithInitials).not.toBeInTheDocument();
  });

  it('should render with the correct data test id', () => {
    render(<Avatar userName="John Doe" dataQaId={dataTestId} />);

    expect(screen.getByTestId(dataTestId)).toBeInTheDocument();
  });

  it('should show loading indicator', () => {
    render(<Avatar userName="John Doe" loading dataQaId={dataTestId} />);

    expect(screen.getByTestId('avatar-loading')).toBeInTheDocument();
  });

  it('should click the profile image', () => {
    const mockFn = jest.fn();
    render(<Avatar userName="John Doe" profileImg={imageSrc} onClick={mockFn} dataQaId={dataTestId} />);

    const avatarContainer = screen.getByTestId(dataTestId);

    userEvent.click(avatarContainer);

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should render a customer icon when no username or image is passed', () => {
    render(<Avatar dataQaId={dataTestId} />);

    expect(screen.getByTestId(`${dataTestId}-customer-icon`)).toBeInTheDocument();
  });

  it('should show a camera icon when avatar is large and editable', () => {
    render(<Avatar dataQaId={dataTestId} editable size={AvatarSize.Large} />);

    expect(screen.getByTestId(`${dataTestId}-camera-icon`)).toBeInTheDocument();
  });

  it('should show a camera icon when avatar is extra large and editable', () => {
    render(<Avatar dataQaId={dataTestId} editable size={AvatarSize.XLarge} />);

    expect(screen.getByTestId(`${dataTestId}-camera-icon`)).toBeInTheDocument();
  });

  it('should not display camera icon', () => {
    render(<Avatar dataQaId={dataTestId} />);

    expect(screen.queryByTestId(`${dataTestId}-camera-icon`)).not.toBeInTheDocument();
  });
});
