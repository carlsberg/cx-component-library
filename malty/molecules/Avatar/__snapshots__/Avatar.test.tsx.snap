// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component matches snapshot 1`] = `
<span
  className="sc-bczRLJ hROJNc"
>
  <div
    className="sc-gKXOVf bGFeFR"
    fontSize="12px"
    loading={false}
    onClick={[Function]}
    size="40px"
  >
    <span
      data-testid="undefined-name"
    >
       
      JD
       
    </span>
  </div>
</span>
`;
